File: cpp_exercises/readability.cpp
#include <iostream>
#include <string>
#include <ctype.h>
#include <cmath>

using namespace std;

int readability(int letter, int word, int sentence);

int main(void)
{
    string text;
    cout << "Texto: ";
    getline(cin, text);

    int textLength = text.size();

    int word = 1;
    int sentence = 0;
    int letter = 0;

    for (int i = 0; i < textLength; i++)
    {
        if (isalpha(text[i]))
        {
            letter++;
        }
        if (isspace(text[i]))
        {
            word++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
    }
    readability(letter, word, sentence);
}

int readability(int letter, int word, int sentence)
{
    float L = ((float)letter / word) * 100;
    float S = ((float)sentence / word) * 100;

    float result = 0.0588 * L - 0.296 * S - 15.8;
    result = round(result);

    if (result <= 1)
    {
        cout << "Before Grade 1";
    }
    else if (result >= 16)
    {
        cout << "Grade 16+";
    }
    else
    {
        cout << "Grade " << result;
    }
    return 0;
}

File: python_exercises/readability.py
import re

text = input("Digite o texto: ")
text_size = len(text)

def readability():
    word_regex = re.findall(r"[\w'-]+", text)
    words = len(word_regex)
    sentences = 0
    letters = 0

    for x in range(0, text_size):
        if text[x].isalpha():
            letters += 1

        if text[x] in [".", "!", "?"]:
            sentences += 1

    calculate_readability(letters, words, sentences)

def calculate_readability(letter, word, sentence):
    L = (letter / word) * 100
    S = (sentence / word) * 100

    result = int(round((0.0588 * L) - (0.296 * S) - 15.8))

    if (result <= 1):
        print("Before Grade 1")

    elif (result >= 16):
        print("Grade 16+")
    
    else:
        print("Grade: {}".format(result))

readability()